File: Java Programming/LinkedlistStack/LinkedlistStackDS.java
public class LinkedlistStackDS {

    LLStackNode head = null;

    public LinkedlistStackDS() {
        System.out.println("\nSuccessful initialization of linkedlist stack.");
    }

    public void push(int data) {
        System.out.println("\nSUCCESS: pushing(" + data + ") into linkedlist stack.");
        if (head == null) {
            head = new LLStackNode(data, null);
        } else {
            LLStackNode newHead = new LLStackNode(data, head);
            head = newHead;
        }
    }

    public void peek() {
        if (head == null) {
            System.out.println("\nERROR: cannot peek from linkedlist stack.");
        } else {
            System.out.println("SUCCESS: head.peek() -> " + head.data);
        }
    }

    public void pop() {
        if (head == null) {
            System.out.println("\nERROR: cannot pop() from empty linkedlist stack.");
        } else {
            System.out.println("\nSUCCESS: popping(" + head.data + ") from linkedlist stack.");
            if (head.getNext() == null) {
                head = null;
            } else {
                head = head.getNext();
            }
        }
    }

    public void printLinkedlistStack() {
        if (head == null) {
            System.out.println("HEAD\n----");
            System.out.println("xxxx");
            System.out.println("----\nNULL");
        } else {
            System.out.println("HEAD\n----");
            LLStackNode currentNode = head;
            while (currentNode != null) {
                System.out.println(currentNode.getData());
                currentNode = currentNode.getNext();
            }
            System.out.println("----\nNULL");
        }
    }
}

File: C Programming/LinkedlistStack/LinkedlistStackDS.c
#include <stdio.h>
#include <stdlib.h>

struct LLStackNode {
    int data;
    struct LLStackNode *next;
} *head = NULL;

void push(int data) {
    printf("\nSUCCESS: pushing(%d) into linkedlist stack.\n", data);
    if (head == NULL) {
        head = (struct LLStackNode*) malloc(sizeof(struct LLStackNode));
        head -> data = data;
        head -> next = NULL;
    } else {
        struct LLStackNode *newHead = (struct LLStackNode*) malloc(sizeof(struct LLStackNode));
        newHead -> data = data;
        newHead -> next = head;
        head = newHead;
    }
}

void peek() {
    if (head == NULL) {
        printf("ERROR: cannot peek from empty linkedlist stack.\n");
    } else {
        printf("SUCCESS: head.peek() --> %d.\n", head -> data);
    }
}

void pop() {
    if (head == NULL) {
        printf("\nERROR: cannot pop() from empty linkedlist stack.\n");
    } else {
        printf("\nSUCCESS: popping(%d) from linkedlist stack.", head -> data);
        if (head -> next == NULL) {
            head = NULL;
        } else {
            head = head -> next;
        }
    }
}

void printLinkedlistStack() {
    if (head == NULL) {
        printf("HEAD\n----\n");
        printf("xxxx\n");
        printf("----\nNULL\n");
    } else {
        printf("HEAD\n----\n");
        struct LLStackNode *currentNode = head;
        while (currentNode != NULL) {
            printf("%d\n", currentNode -> data);
            currentNode = currentNode -> next;
        }
        printf("----\nNULL\n");
    }
}

int main(int argc, char* argv[]) {
    printf("LinkedlistStack.c\n");
    printf("-----------------\n");
    printLinkedlistStack();
    push(3);
    printLinkedlistStack();
    peek();
    push(22);
    printLinkedlistStack();
    peek();
    push(7);
    printLinkedlistStack();
    peek();
    push(81);
    printLinkedlistStack();
    peek();
    pop();
    printLinkedlistStack();
    peek();
    pop();
    printLinkedlistStack();
    peek();
    pop();
    printLinkedlistStack();
    peek();
    pop();
    printLinkedlistStack();
    peek();
    pop(); // ERROR
    printLinkedlistStack();
    peek();
    return EXIT_SUCCESS;
}